nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,438 @@
+    "device__location__parent__parent__id": "6695809c-b33b-4f12-b0de-a4969000434d",
+    "device__location__parent__parent__id__in": ["6695809c-b33b-4f12-b0de-a4969000434d", "e51d07bb-3fcf-4306-9d87-6b1ff6dd6378"]
+}
+```
+
+### Other Data Model Specific ObjectPermission e.g. Note
+
+The old `constraints` field for a `Site`/`Region` related data model's (e.g. `Note`) `ObjectPermission` instance might look like this:
+
+```json
+{
+    "assigned_object_type": "dcim.site",
+    "assigned_object_id": "932d94ee-5571-40a0-903f-4274fcfbed32",
+    "assigned_object_id__in": ["932d94ee-5571-40a0-903f-4274fcfbed32", "e383db9a-dd55-464d-9e56-2f18bc03b32c"]
+}
+```
+
+To modify the data correctly, we need to:
+    1. Replace all occurrences of "dcim.site" and/or "dcim.region" with "dcim.location" in the **Value** portion (after ":") of the `assigned_object_type` Key.
+    2. Since Nautobot carries over the UUIDs of the old `Site`/`Region` instances when creating the new "Site"/"Region" type `Location` instances, we **do not** need to change the UUID values in the `assigned_object_id` and `assigned_object_id__in` Keys.
+
+The updated JSON data might look like this:
+
+```json
+{
+    "assigned_object_type": "dcim.location",
+    "assigned_object_id": "932d94ee-5571-40a0-903f-4274fcfbed32",
+    "assigned_object_id__in": ["932d94ee-5571-40a0-903f-4274fcfbed32", "e383db9a-dd55-464d-9e56-2f18bc03b32c"]
+}
+```
+
+## Region and Site Related Data Model Migration Guide For Existing Nautobot App installations
+
+In Nautobot 2.0.0, all the `Region` and `Site` related data models are being migrated to use `Location`. Below is a comprehensive guide for Nautobot App developers to migrate their `Region` and `Site` related data models to `Location`.
+
+We will be using `ExampleModel` as a relatively simple and hands-on example throughout this guide to better your understanding of the migration process.
+
+!!!important  
+    Before you follow the guide, please make sure that these operations are completed:  
+        1. Make sure your working Nautobot is on Version 1.5.x with baseline migrations all run.  
+        2. Stop Nautobot Server.  
+        3. Update installed version of Nautobot to 2.0.0.  
+        4. Run `nautobot-server migrate dcim 0030_migrate_region_and_site_data_to_locations`. (This step will ensure that `("dcim", "0030_migrate_region_and_site_data_to_locations")` is applied to your Nautobot instance and that `("dcim", "0034_remove_region_and_site")` is **not** applied.)  
+    After you complete those operations, follow the guide below for each of your installed apps to:  
+        1. Make all necessary migration files for each app.  
+        2. Run `nautobot-server migrate [app_name]` to apply those migration files to each app.  
+        3. Finally, Start Nautobot Server after **all** the migration files are applied and **all** your affected apps are updated.  
```

Comment:
This is a lot of information stuffed into an admonition.

It would be better as an h3 section with an admonition at the top that is more than just "important". Probably "warning" so that it's orange and stands out. 

It would be good for this admonition to say something like "You must perform these steps." It needs to be very explicit that if you just run the migrations without stopping at `0030` that you absolutely will nuke your data.

Additionally, there should be an extra step after `2. Stop Nautobot Server` that tells the reader to make a database backup. We don't have any documentation on how to backup data. The closest thing we have is a section on "Export data from PostgreSQL" in the "Migrating from PostgreSQL to MySQL" document.

The suggested changes are not complete. Just to give an example.

```suggestion
### Before you Begin 

!!! warning
    You **must** perform these steps before proceeding. Failing to follow them properly could result in data loss. Always backup your database before performing any migration steps.
    
Before you follow the guide, please make sure that these operations are completed:  

1. Make sure your working Nautobot is on Version 1.5.x with baseline migrations all run.  
2. Stop Nautobot Server.  
3. Create a backup of your Nautobot database.
4. Update installed version of Nautobot to 2.0.0.  
5. Run `nautobot-server migrate dcim 0030_migrate_region_and_site_data_to_locations`. (This step will ensure that `("dcim", "0030_migrate_region_and_site_data_to_locations")` is applied to your Nautobot instance and that `("dcim", "0034_remove_region_and_site")` is **not** applied.)  

After you complete those operations, follow the guide below for each of your installed apps to:  

1. Make all necessary migration files for each app.  
2. Run `nautobot-server migrate [app_name]` to apply those migration files to each app.  
3. Finally, Start Nautobot Server after **all** the migration files are applied and **all** your affected apps are updated.  
```

Reply:
Is `nautobot-server dumpdata` something you can use to back up your data?

Reply:
Short answer is yes. It is a solution that works across any database backend.